docs/data/material/customization/overriding-component-structure/overriding-component-structure.md

```diff
@@ -0,0 +1,108 @@
+
+For example, the code snippet below shows how to add a custom `data-testid` to the popper slot of the [Autocomplete](/material-ui/react-autocomplete/) component:
+
+```jsx
+<Autocomplete slotProps={{ popper: { 'data-testid': 'my-popper' } }} />
+```
+
+All additional props placed on the primary component are also propagated into the root slot (just as if they were placed in `slotProps.root`).
+These two examples are equivalent:
+
+```jsx
+<Autocomplete id="badge1">
+```
+
+```jsx
+<Autocomplete slotProps={{ root: { id: 'badge1' } }}>
+```
+
+:::warning
+If both `slotProps.root` and additional props have the same keys but different values, the `slotProps.root` props will take precedence.
+This does not apply to classes or the `style` prop—they will be merged instead.
+:::
+
+## Best practices
+
+Use `component` or `slotProps.{slot}.component` prop to override the element by preserving the styles of the slot.
+
+Use `slots` prop to replace the slot's styles and functionality with your custom component.
+
+Overriding with `component` lets you apply the attributes of that element directly to the root.
+For instance, if you override the Button's root with an `<li>` tag, you can add the `<li>` attribute `value` directly to the component.
+If you did the same with `slots.root`, you would need to place this attribute on the `slotProps.root` object in order to avoid a TypeScript error.
+
+Be mindful of your rendered DOM structure when overriding the slots of more complex components.
+You can easily break the rules of semantic and accessible HTML if you deviate too far from the default structure—for instance, by unintentionally nesting block-level elements inside of inline elements.
+
+## History of the pattern
+
+Prior to Material UI v6, majority of the components lacked the ability to override their structure.
+Some components had `*Props` props that allowed you to pass props to a specific slot, but this was not a consistent pattern across the library.
+
+In v6, those props were deprecated in favor of the `slots` and `slotProps` props, which allow for more granular control over the structure of a component and make the API more consistent across the library.
+
+The summary of the benefits of the new pattern is as follows:
+
+- The `slotProps` provided to [theme default props](/material-ui/customization/theme-components/#theme-default-props) are merged with the instance `slotProps` prop. This is useful for building custom themes or use Material UI as a design system component library.
+- Most components have a `slots` prop that allows developers to replace the entire slot with a custom one. For example, developers can remove the popper behavior from the Autocomplete component by replacing `popper` slot with a `div` as [shown above](#the-slots-prop).
+- Integrate better with [Tailwind CSS](https://tailwindcss.com/) because developers can provide utility classes to the slots through `slotProps.*.className`.
```

Comment:
```suggestion
- Offers better integration with [Tailwind CSS](https://tailwindcss.com/) because developers can provide utility classes to the slots through `slotProps.*.className`.
```